smithy-model/src/main/java/software/amazon/smithy/model/validation/ValidationEventDecorator.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.smithy.model.validation;
+
+/**
+ * Validation events decorators take validations events and transform them by adding more contextual information,
+ * usually adding a hint to let the user know what can it be done to solve the issue. This might add context specific
+ * information that is not relevant for all cases such as links to internal knowledge sites or explicit instructions
+ * relevant only to the context where Smithy is being used.
+ */
+@FunctionalInterface
+public interface ValidationEventDecorator {
```

Comment:
Should this have a separate predicate method? That would enable a separation of concerns, make `decorate` only do actual decoration.

Reply:
Makes sense, I changed it accordingly by adding a new `canDecorate(ValidationEvent)` method.